src/test/java/com/stripe/functional/BankAccountTest.java

```diff
@@ -0,0 +1,166 @@
+package com.stripe.functional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import com.stripe.BaseStripeTest;
+import com.stripe.exception.StripeException;
+import com.stripe.model.BankAccount;
+import com.stripe.model.Customer;
+import com.stripe.model.DeletedBankAccount;
+import com.stripe.model.ExternalAccount;
+import com.stripe.model.ExternalAccountCollection;
+import com.stripe.net.APIResource;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Test;
+
+public class BankAccountTest extends BaseStripeTest {
+  public static final String CUSTOMER_ID = "cus_123";
+  public static final String BANK_ACCOUNT_ID = "ba_123";
+
+  private static BankAccount getBankAccountFixture(Customer customer) throws StripeException {
+    final BankAccount bankAccount = (BankAccount) customer.getSources().retrieve(BANK_ACCOUNT_ID);
+    resetNetworkSpy();
+    // stripe-mock returns a bank account that is attached to an account rather than a customer,
+    // so manually fix that. Hopefully this will no longer be necessary in the future.
+    bankAccount.setAccount(null);
+    bankAccount.setCustomer(customer.getId());
+
+    return bankAccount;
+  }
+
+  @Test
+  public void testCreate() throws StripeException {
+    final Customer customer = Customer.retrieve(CUSTOMER_ID);
+
+    Map<String, Object> params = new HashMap<String, Object>();
+    params.put("source", "btok_123");
+
+    final BankAccount bankAccount = (BankAccount) customer.getSources().create(params);
+
+    assertNotNull(bankAccount);
+    verifyRequest(
+        APIResource.RequestMethod.POST,
+        String.format("/v1/customers/%s/sources", customer.getId()),
+        params
+    );
+  }
+
+  @Test
+  public void testRetrieve() throws StripeException {
+    final Customer customer = Customer.retrieve(CUSTOMER_ID);
```

Comment:
why do we have `final` here and not on any of the other examples? Should we fix them all? (annoying I know but consistency is king)

Reply:
Immutability is good, so ideally all variables that can be declared as `final` _should_ be. I think there's a linter rule that require that variables whose declaration is more than X lines away from their first use be declared as final, which is what prompted me to start using it in some places, and then thought that we might as well use it everywhere.

In practice, I don't think it's worth stressing about too much.